nanome/api/schemas/structure_schemas.py

```diff
@@ -168,3 +170,72 @@ def make_workspace(self, data, **kwargs):
         new_obj = structure.Workspace()
         init_object(new_obj, data)
         return new_obj
+
+
+class StructureSchema(Schema):
+
+    structure_schema_map = {
+        structure.Atom: AtomSchema(),
+        structure.Bond: BondSchema(),
+        structure.Residue: ResidueSchema(),
+        structure.Chain: ChainSchema(),
+        structure.Molecule: MoleculeSchema(),
+        structure.Complex: ComplexSchema(),
+    }
+
+    def dump(self, obj, *args, **kwargs):
+        obj_class = obj.__class__
+        schema = self.structure_schema_map[obj_class]
+        dump_data = schema.dump(obj, *args, **kwargs)
+        return dump_data
+
+    def load(self, data, *args, **kwargs):
+        correct_schema = self.determine_structure_schema(data)
+        return correct_schema.load(data, *args, **kwargs)
+
+    def determine_structure_schema(self, data):
+        """Use unique fields to determine schema to use for provided data."""
+        complex_fields = list(ComplexSchema._declared_fields.keys())
+        molecule_fields = list(MoleculeSchema._declared_fields.keys())
+        chain_fields = list(ChainSchema._declared_fields.keys())
+        residue_fields = list(ResidueSchema._declared_fields.keys())
+        atom_fields = list(AtomSchema._declared_fields.keys())
+        bond_fields = list(BondSchema._declared_fields.keys())
+        unique_complex_fields = [
+            field for field in complex_fields
+            if field not in set(molecule_fields + chain_fields + residue_fields + atom_fields + bond_fields)]
+        unique_molecule_fields = [
+            field for field in molecule_fields
+            if field not in set(complex_fields + chain_fields + residue_fields + atom_fields + bond_fields)]
+        unique_chain_fields = [
+            field for field in chain_fields
+            if field not in set(complex_fields + molecule_fields + residue_fields + atom_fields + bond_fields)]
+        unique_residue_fields = [
+            field for field in residue_fields
+            if field not in set(complex_fields + molecule_fields + chain_fields + atom_fields + bond_fields)]
+        unique_bond_fields = [
+            field for field in bond_fields
+            if field not in set(complex_fields + molecule_fields + chain_fields + residue_fields + atom_fields)]
+        unique_atom_fields = [
+            field for field in atom_fields
+            if field not in set(complex_fields + molecule_fields + chain_fields + residue_fields + bond_fields)]
```

Comment:
You can use set difference here instead of all these list comprehensions:

```
unique_complex_fields = set(complex_fields) - set(molecule_fields + chain_fields + residue_fields + atom_fields + bond_fields)
```